Data/models/RSSFeedSource.swift

```diff
@@ -0,0 +1,45 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import CoreData
+
+public final class RSSFeedSource: NSManagedObject, CRUD {
+    @NSManaged public var title: String?
+    @NSManaged public var feedUrl: String
+    
+    public class func get(with feedUrl: String) -> RSSFeedSource? {
+        let predicate = NSPredicate(format: "\(#keyPath(RSSFeedSource.feedUrl)) == %@", feedUrl)
+        return first(where: predicate, context: DataController.viewContext)
+    }
+    
+    public class func all() -> [RSSFeedSource] {
+        all() ?? []
+    }
+    
+    public class func delete(with feedUrl: String) {
+        let context = DataController.viewContext
+        if let item = get(with: feedUrl) {
+            item.delete(context: .existing(context))
```

Comment:
Do we have to delete it on main thread/context? do we rely on it somewhere in code?

Reply:
UI doesn't update correctly, we delete & refetch, but since its backgrounded it doesn't fetch correctly. Easier for such a small entity to just use main thread by default